### PR TITLE
Add online feature retrieval integration test using the universal repo

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/tests/integration/feature_repos/universal/data_sources/file.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def create_data_sources(
             event_timestamp_column=event_timestamp_column,
             created_timestamp_column=created_timestamp_column,
             date_partition_column="",
-            field_mapping=field_mapping or {"ts_1": "ts", "id": "driver_id"},
```
Comment (adchia) [inline sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:38]: intentional removal?

File: sdk/python/tests/integration/feature_repos/universal/feature_views.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def create_customer_daily_profile_feature_view(source):
             Feature(name="current_balance", dtype=ValueType.FLOAT),
             Feature(name="avg_passenger_count", dtype=ValueType.FLOAT),
             Feature(name="lifetime_trip_count", dtype=ValueType.INT32),
-            Feature(name="avg_daily_trips", dtype=ValueType.INT32),
+            # Feature(name="avg_daily_trips", dtype=ValueType.INT32),
```
Comment (adchia) [inline sdk/python/tests/integration/feature_repos/universal/feature_views.py:42]: intentional comment?

File: sdk/python/tests/integration/online_store/test_universal_online.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+import pandas as pd
+
+from tests.integration.feature_repos.test_repo_configuration import (
+    Environment,
+    parametrize_online_test,
+)
+
+
+@parametrize_online_test
```
Comment (adchia) [inline sdk/python/tests/integration/online_store/test_universal_online.py:12]: is the intention to add more tests here? the original test seems to also e.g. test the TTL of the registry and to_df

File: sdk/python/tests/integration/online_store/test_universal_online.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    driver_rows = driver_df[driver_df["driver_id"] == entity_row["driver"]]
+    latest_driver_row: pd.DataFrame = driver_rows.loc[
+        driver_rows["event_timestamp"].idxmax()
+    ].to_dict()
+    customer_rows = customer_df[customer_df["customer_id"] == entity_row["customer_id"]]
+    latest_customer_row = customer_rows.loc[
+        customer_rows["event_timestamp"].idxmax()
+    ].to_dict()
+
+    latest_customer_row.update(latest_driver_row)
```
Comment (adchia) [inline sdk/python/tests/integration/online_store/test_universal_online.py:114]: is this update call what you really mean? i'd guess that you actually want to use pd.concat on the customer + driver features? 

this iiuc is finding any overlapping feature names and using the driver values.... which should mainly just be the timestamp?

File: sdk/python/tests/integration/feature_repos/universal/data_sources/file.py (hunk 1 above)
Comment (achals) [reply sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:38]: Yes, intentional to be consistent with the other sources

File: sdk/python/tests/integration/feature_repos/universal/feature_views.py (hunk 2 above)
Comment (achals) [reply sdk/python/tests/integration/feature_repos/universal/feature_views.py:42]: Yes, this feature doesnt actually exist in the data backed by this view.

File: sdk/python/tests/integration/online_store/test_universal_online.py (hunk 3 above)
Comment (achals) [reply sdk/python/tests/integration/online_store/test_universal_online.py:12]: I'll add the to_df tests and registry tests as well

File: sdk/python/tests/integration/online_store/test_universal_online.py (hunk 4 above)
Comment (achals) [reply sdk/python/tests/integration/online_store/test_universal_online.py:114]: No, the update is just merging the two dicts which is functionally similar to the pd.concat. the merged dict contains the latest feature values for both drivers and customers.

Comment (adchia) [reply sdk/python/tests/integration/online_store/test_universal_online.py:114]: misread and thought these were dfs. resolving

File: sdk/python/tests/integration/feature_repos/universal/feature_views.py (hunk 2 above)
Comment (adchia) [reply sdk/python/tests/integration/feature_repos/universal/feature_views.py:42]: can we just remove it then?

**Review by adchia — CHANGES_REQUESTED** [review]
/lgtm

File: sdk/python/tests/integration/online_store/test_universal_online.py (hunk 3 above)
Comment (achals) [reply sdk/python/tests/integration/online_store/test_universal_online.py:12]: Going to add this in a different PR.

**Review by adchia — APPROVED** [review]

**Pull request merged**